tests/Doctrine/Tests/ORM/Functional/SQLFilterTest.php

```diff
@@ -328,6 +328,36 @@ public function testSQLFilterSetParameterInfersType(): void
         $this->assertEquals("'en'", $filter->getParameter('locale'));
     }
 
+    public function testSQLFilterSetArrayParameterInfersType(): void
+    {
+        // Setup mock connection
+        $conn = $this->getMockConnection();
+        $conn->expects($this->at(0))
+            ->method('quote')
+            ->with($this->equalTo('en'))
+            ->will($this->returnValue("'en'"));
+        $conn->expects($this->at(1))
+            ->method('quote')
+            ->with($this->equalTo('es'))
+            ->will($this->returnValue("'es'"));
```

Comment:
Please use `returnValueMap()` here. The invocation order of the mock isn't relevant and is covered in the resulting assertion anyways.